Resource.js

```diff
@@ -393,18 +403,36 @@ module.exports = function(app, route, modelName, model) {
     put: function(options) {
       this.methods.push('put');
       this.register(app, 'put', this.route + '/:' + this.name + 'Id', function(req, res, next) {
-        if (req.skipResource) { return next(); }
+        if (req.skipResource) {
+          debug.put('Skipping Resource');
+          return next();
+        }
+
+        // Remove __v field
+        if (req.body.hasOwnProperty('__v')) {
+          delete req.body.__v;
+        }
+
+        debug.put('Update: ' + JSON.stringify(req.body));
         var query = req.modelQuery || this.model;
-        query.findOne({'_id': req.params[this.name + 'Id']}, function(err, item) {
-          if (err) return this.setResponse(res, {status: 500, error: err}, next);
-          if (!item) return this.setResponse(res, {status: 404}, next);
-          if (req.body.hasOwnProperty('__v')) { delete req.body.__v; }
-          item.set(req.body);
-          item.save(function (err, item) {
-            if (err) return this.setResponse(res, {status: 400, error: err}, next);
+        query.findOneAndUpdate(
+          {_id: req.params[this.name + 'Id']},
+          {$set: req.body},
+          {new: true},
```

Comment:
This has caused significant issues since findOneAndUpdate bypasses setters, validation and middleware. See http://stackoverflow.com/questions/18040315/why-arent-defaults-setters-validators-and-middleware-not-applied-for-findonea